translate/src/modules/machinery/components/source/GoogleTranslation.tsx

```diff
@@ -1,27 +1,156 @@
-import React from 'react';
+import React, { useState, useRef, useContext, useEffect } from 'react';
 import { Localized } from '@fluent/react';
+import type { MachineryTranslation } from '~/api/machinery';
+import { fetchGPTTransform } from '~/api/machinery';
+import { Locale } from '~/context/Locale';
+import { logUXAction } from '~/api/uxaction';
+
+type Props = {
+  translation: MachineryTranslation;
+  onLLMTranslationChange: (llmTranslation: string) => void;
+};
 
 /**
  * Show the translation source from Google Translate.
  */
-export function GoogleTranslation(): React.ReactElement<'li'> {
+
+export function GoogleTranslation({
+  translation,
+  onLLMTranslationChange,
+}: Props): React.ReactElement<'li'> {
+  const [isDropdownOpen, setDropdownOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState('');
+  const [showOriginalOption, setShowOriginalOption] = useState(false);
+  const [currentTranslation, setCurrentTranslation] = useState(
+    translation.translation,
+  );
+  const dropdownRef = useRef<HTMLLIElement>(null);
+  const locale = useContext(Locale);
+  const [llmTranslation, setLLMTranslation] = useState('');
+
+  const toggleDropdown = (ev: React.MouseEvent) => {
+    ev.stopPropagation();
```

Comment:
Why is this line needed? Same applies to all `stopPropagation()` calls below, I'm pretty sure none of them have any effect.

Reply:
This is needed as well as the call in the `handleOptionClick` function. Without these calls, clicking on the dropdown menu causes the text in the machinery tab to be copied into the editor (and the dropdown menu does not open).